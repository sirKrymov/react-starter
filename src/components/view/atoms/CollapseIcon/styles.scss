@import '../../../../styles/configs';

.collapse-icon {
  &--open {
    transform: rotate(180deg);
  }
}
