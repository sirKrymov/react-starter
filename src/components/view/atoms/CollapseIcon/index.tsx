// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { TiArrowSortedDown } from 'react-icons/ti';
import cn from 'classnames';

import { ReactIcon } from 'components/view/atoms/ReactIcon';

import './styles.scss';

interface IProps {
  onClick?(): void;
  open?: boolean;
  id: string;
}

export function CollapseIcon({ onClick, open, id }: IProps): ReactElement {
  const collapseIconClass = cn({
    'collapse-icon': true,
    'collapse-icon--open': open
  });

  return (
    <div className={collapseIconClass} onClick={onClick} id={id}>
      <ReactIcon size="md" color="on-primary">
        <TiArrowSortedDown />
      </ReactIcon>
    </div>
  );
}
