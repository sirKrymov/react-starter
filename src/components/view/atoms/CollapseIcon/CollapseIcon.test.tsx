import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import { CollapseIcon } from '../CollapseIcon';

describe('<CollapseIcon> atom component', () => {
  const handleClick = sinon.spy();
  let shallowComponent: ShallowWrapper;

  const createComponent = (props?: Record<string, any>) => {
    shallowComponent = shallow(<CollapseIcon id="collapse-icon" {...props} />);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });

  describe('WHEN the component have additional properties', () => {
    it('WHEN it should be handle click logic', () => {
      createComponent({ onClick: handleClick });

      expect(shallowComponent.find('div.collapse-icon')).toHaveLength(1);
      shallowComponent.find('div').simulate('click');
      expect(handleClick).toHaveProperty('callCount', 1);
    });

    it('WHEN it should be with open property', () => {
      createComponent({ open: true });

      expect(shallowComponent.find('div.collapse-icon--open')).toHaveLength(1);
    });
  });
});
