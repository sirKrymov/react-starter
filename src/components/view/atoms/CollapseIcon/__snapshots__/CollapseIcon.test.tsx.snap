// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapseIcon> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="collapse-icon"
  id="collapse-icon"
>
  <ReactIcon
    color="on-primary"
    size="md"
  >
    <TiArrowSortedDown />
  </ReactIcon>
</div>
`;
